[W10-1] README

qjqqyy opened the pull request:
@qjqqyy @rlrh @thomastanck @epicfailname

langyanbin0314 commented on docs/AboutUs.adoc:
```diff
@@ -5,52 +5,43 @@
-image::yijinl.jpg[width="150", align="left"]
-{empty}[http://github.com/yijinl[github]] [<<johndoe#, portfolio>>]
+=== Thomas Tan
+image::thomastanck.png[width="150", align="left"]
```
should show your own face

langyanbin0314 commented on docs/AboutUs.adoc:
```diff
@@ -5,52 +5,43 @@
 AddressBook - Level 4 was developed by the https://se-edu.github.io/docs/Team.html[se-edu] team. +
-_{The dummy content given below serves as a placeholder to be used by future forks of the project.}_ +
-{empty} +
+
```
please specify each member's responsibilities. e.g. who does which component

langyanbin0314 commented on README.adoc:
```diff
@@ -14,26 +14,28 @@ ifndef::env-github[]
-** A more sophisticated GUI that includes a list  panel and an in-built Browser.
-** More test cases, including automated GUI testing.
-** Support for _Build Automation_ using Gradle and for _Continuous Integration_ using Travis CI.
+* README helps you discover, curate and read more content you care about in less time.
```
good job on the mock UI

langyanbin0314 commented on docs/UserGuide.adoc:
```diff
@@ -12,147 +12,599 @@ ifdef::env-github[]
 ====
 
-=== Viewing help : `help`
+== Viewing help : `help`
```
should label which features have been done and which are 'coming in v2.0'

langyanbin0314 commented on docs/DeveloperGuide.adoc:
```diff
@@ -826,36 +839,156 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`* * *` |user |delete a person |remove entries that I no longer need
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|Priority |As a ... |I want to ... |So that I can...
```
it would be better you can group and order the User stories by priority

langyanbin0314 commented on docs/DeveloperGuide.adoc:
```diff
@@ -881,6 +1117,8 @@ _{More to be added}_
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `9` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  A heavy user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
should quantify 'heavy' and 'above average type speed' i.e how many words per min

langyanbin0314 commented:
generally well done

qjqqyy committed:
Merge pull request #181 from CS2103-AY1819S2-W10-1/exit-cleanly

Network: clean up AsyncHttpClient on exit

qjqqyy committed:
rename feed context add -> addindex

qjqqyy committed:
lift add command

qjqqyy committed:
lift subscribe

qjqqyy committed:
UG: Move add and subscribe into non-contextual

also link to them from their respective contexts

qjqqyy committed:
UG: while here document list context refresh

it's not called redownload

qjqqyy committed:
move select from EBP into the inherited parsers

qjqqyy committed:
UG: document archives/select

qjqqyy committed:
Parsers: add a couple tests

qjqqyy committed:
Bing: add help message

qjqqyy committed:
FeedUtil: take in Set of Tags

qjqqyy committed:
feeds/{sub,refresh}: tag entries

when subscribing to or refreshing a feed we now tag the related entries
with whatever the feed is also tagged with

qjqqyy committed:
Merge pull request #183 from CS2103-AY1819S2-W10-1/feed-tags

Tag entries when subscribing to a feed

rlrh committed:
Initial rewrite

epicfailname committed:
Merge branch 'master' into update-dg

epicfailname committed:
Merge branch 'master' into update-ug

epicfailname committed:
Merge pull request #167 from CS2103-AY1819S2-W10-1/update-ug

 Fix find command documentation

qjqqyy committed:
Merge branch 'master' into move-commands-around

qjqqyy committed:
Merge pull request #182 from CS2103-AY1819S2-W10-1/move-commands-around

Promote subscribe and add to all contexts

rlrh committed:
Debug

epicfailname committed:
Merge branch 'master' into update-dg

epicfailname committed:
Merge pull request #185 from CS2103-AY1819S2-W10-1/update-dg

Add diagrams for Context-Specific Commands Implementation

epicfailname committed:
Merge branch 'master' into final-final-final-branch

rlrh committed:
Move getArticle implementation from Model to ArticleStorage

thomastanck committed:
Merge branch 'master' into 3-column

epicfailname committed:
Change link to PPP to match requirements

rlrh committed:
Working implementation

epicfailname committed:
First draft

epicfailname committed:
Second draft

rlrh committed:
Clean up and comment code, update mocks

rlrh committed:
Refactor internal URLs into constant list

rlrh committed:
Simplify code on assumption that there are no relative links

rlrh committed:
Debug and clean up code, add more comments

rlrh committed:
Remove asserts in BrowserPanel

rlrh committed:
Merge branch 'master' into rewrite-browser-panel

rlrh committed:
Fix reader view async bug by removing Platform.runLater

thomastanck committed:
Extract method from AbsoluteUrlDocumentConverter#convert

thomastanck committed:
Implement CSS url conversion

thomastanck committed:
Don't auto download articles on select when not in list context

thomastanck committed:
Add some logging messages to catch issues like #187 earlier

epicfailname committed:
Add more system tests

epicfailname committed:
Remove unused variables

epicfailname committed:
Fix indentation

thomastanck committed:
Refactor AddCommand's duplicate entry exception to own class

thomastanck committed:
Add addall command

rlrh committed:
Change icon

rlrh committed:
Merge branch '3-column' of https://github.com/rlrh/main into 3-column

rlrh committed:
Merge pull request #160 from rlrh/3-column

Update UI to be more user-friendly

thomastanck committed:
Add tests for AddIndexCommand

thomastanck committed:
Add test for SubscribeCommandParser

thomastanck committed:
Add more test cases to ParserUtilTest#parseIndex*

thomastanck committed:
Add tests for commands which accept an index

thomastanck committed:
Add missing integration tests for EntryBook*ParserTest

thomastanck committed:
Add search commands' parser tests

epicfailname committed:
Merge branch 'master' into final-final-final-branch

thomastanck committed:
Add DateUtilTest

thomastanck committed:
Add some random ModelManager tests

thomastanck committed:
Rename ClearCommand to ClearListCommand

thomastanck committed:
Add ClearArchivesCommand and tests

thomastanck committed:
Add ClearFeedsCommand and tests

thomastanck committed:
Make ParserUtil abstract

epicfailname committed:
Merge pull request #175 from CS2103-AY1819S2-W10-1/final-final-final-branch

Entry Management System: Add tests for Archive and Unarchive

thomastanck committed:
Add archiveall command

thomastanck committed:
Add test for refresh command

thomastanck committed:
Add refreshall command

thomastanck committed:
Add unarchiveall command

thomastanck committed:
Fix typo in test: setup -> setUp

thomastanck committed:
Add tests for logic's context switching callbacks

thomastanck committed:
Remove unused facade in Logic

thomastanck committed:
Remove unused code

thomastanck committed:
Add refreshall for feeds

thomastanck committed:
Improve logging messages (and upgrade logging severity to WARNING)

thomastanck committed:
Add test for DeleteArchiveEntryCommand

thomastanck committed:
Merge pull request #188 from thomastanck/select-downloads-only-in-list

Don't auto download articles on select when not in list context

thomastanck committed:
Merge branch 'master' into more-tests

thomastanck committed:
Make {Add,Archive,Unarchive}AllCommandTest more precise

thomastanck committed:
Merge pull request #191 from thomastanck/more-tests

Add tests for AddIndexCommand

thomastanck committed:
Test has{List,Archives,Feeds,}Entry separately

thomastanck committed:
Split up delete*Entry tests

thomastanck committed:
Split up setSelectedEntry tests for each context

thomastanck committed:
Split up deleteEntry tests for each context

thomastanck committed:
Merge remote-tracking branch 'origin/master' into all-commands

thomastanck committed:
Merge branch 'master' into parser-tests

thomastanck committed:
Add missing parseCommand for refreshall in feeds context

thomastanck committed:
Add tests for trivially successful refreshall command

thomastanck committed:
Remove unused constructors for DuplicateEntryCommandException

thomastanck committed:
Fix typo for one of the tests

tests all has*Entry rather than just hasListEntry

thomastanck committed:
Merge pull request #192 from thomastanck/parser-tests

Add tests for parsers

thomastanck committed:
Merge branch 'master' into promote-clear-command

thomastanck committed:
Merge remote-tracking branch 'origin/master' into all-commands

thomastanck committed:
Merge pull request #195 from thomastanck/promote-clear-command

Promote clear command to archives and feeds context

thomastanck committed:
Merge branch 'master' into random-tests

thomastanck committed:
Merge pull request #193 from thomastanck/random-tests

Add tests and remove code to increase coverage

thomastanck committed:
Merge branch 'master' into all-commands

thomastanck committed:
Merge pull request #190 from thomastanck/all-commands

Add "all" variants of commands

thomastanck committed:
Merge branch 'master' into rewrite-browser-panel

thomastanck committed:
Merge branch 'master' into absolutize-css-links

rlrh committed:
Add GoogleNewsCommandParserTest

rlrh committed:
Add tests in StringUtilTest

rlrh committed:
Add test for displaying previous page after switching view modes in BrowserPanelTest

rlrh committed:
Update README libraries used, marketing info

rlrh committed:
Change from getArticle in Storage to getArticleAsOptionalString in Network

thomastanck committed:
Autofill feed tag when subscribing to a feed initially

thomastanck committed:
Merge branch 'master' into update-readme

rlrh committed:
Display loading progress bar

thomastanck committed:
Fix failing test

thomastanck committed:
Get rid of startup sample data

epicfailname committed:
Add pictures to UG find and archive command

rlrh committed:
Refactor BrowserPanel - reorganize, rename to be clearer

rlrh committed:
Update Network and NetworkTest

rlrh committed:
Merge pull request #197 from rlrh/update-readme

Update README.adoc libraries used, marketing info

rlrh committed:
Fix checkstyle error

epicfailname committed:
Reformat example command sequences

epicfailname committed:
Resize a picture

rlrh committed:
Change team info, add rlrh portfolio link

epicfailname committed:
Underline Format and Examples subheader

rlrh committed:
Merge branch 'master' into rewrite-browser-panel

epicfailname committed:
Merge branch 'master' into update-ug

thomastanck committed:
Network: Follow redirects (up to 15 times)

thomastanck committed:
Add test to check that redirects work

epicfailname committed:
Add Context section

thomastanck committed:
Improve network error message when visiting nonexistent website

epicfailname committed:
Merge branch 'update-ug' of github.com:CS2103-AY1819S2-W10-1/main into update-ug

thomastanck committed:
Merge branch 'master' into network-follow-redirects

thomastanck committed:
Merge branch 'master' into absolutize-css-links

rlrh committed:
Merge pull request #184 from rlrh/rewrite-browser-panel

Improve Reader View performance, rewrite BrowserPanel

thomastanck committed:
Add missing refreshall command to list context

thomastanck committed:
Merge branch 'master' into add-missing-refreshall-command

epicfailname committed:
Add images for context switching commands

thomastanck committed:
Network: Refactor magic number to constant DEFAULT_NUM_REDIRECTS

epicfailname committed:
Merge branch 'master' into update-ug

thomastanck committed:
Merge remote-tracking branch 'origin/master' into network-follow-redirects

thomastanck committed:
Refactor Network#fetchAsOptionalString

thomastanck committed:
Merge branch 'master' into absolutize-css-links

epicfailname committed:
Merge branch 'master' into add-ppp-jon

thomastanck committed:
Merge pull request #186 from thomastanck/absolutize-css-links

Implement converter for CSS links

thomastanck committed:
Merge branch 'master' into add-ppp-jon

thomastanck committed:
Remove unused function

rlrh committed:
Merge branch 'master' into update-ug

epicfailname committed:
Merge pull request #205 from CS2103-AY1819S2-W10-1/add-ppp-jon

Update PPP for Jonathan

epicfailname committed:
Merge branch 'master' into update-ug

rlrh committed:
Update view mode UG section with images and more examples

rlrh committed:
Merge branch 'update-ug' of https://github.com/CS2103-AY1819S2-W10-1/main into update-ug

thomastanck committed:
Merge branch 'master' into subscribe-default-tag

thomastanck committed:
Merge pull request #198 from thomastanck/subscribe-default-tag

Autofill feed tag when subscribing to a feed initially

thomastanck committed:
Tweak test: check strings are contained in exception instead of exact match

thomastanck committed:
Merge remote-tracking branch 'origin/master' into network-follow-redirects

qjqqyy committed:
test/TypicalEntries: refactor base urls out

thomastanck committed:
Merge pull request #203 from thomastanck/network-follow-redirects

Network: Follow redirects (up to 15 times)

qjqqyy committed:
test/FeedCommand: use URLs from TypicalEntries

thomastanck committed:
Merge branch 'master' into update-ug

qjqqyy committed:
test/FeedUtil: remove makeRssFromTriple

epicfailname committed:
Merge pull request #202 from CS2103-AY1819S2-W10-1/update-ug

UserGuide: Add images and change formatting

qjqqyy committed:
test: move animereviews feed to TypicalEntries

also use TypicalEntries in FeedUtilTest

qjqqyy committed:
test/FeedCommand: use NXDomain url from TypicalEntries

promote magic string "this.web.site.does.not.exist.definitely" to
TypicalEntries.NONEXISTENT_DOMAIN

qjqqyy committed:
test/FeedCommand: remove imported unused Exception

rlrh committed:
Credit se-edu team in AboutUs

qjqqyy committed:
test: lift ModelStub out

thomastanck committed:
Scroll list panel to any changes made to the currently viewed list

thomastanck committed:
Merge branch 'master' into scroll-to-add

thomastanck committed:
Merge branch 'master' into add-missing-refreshall-command

thomastanck committed:
Improve logging message

thomastanck committed:
Merge pull request #207 from thomastanck/scroll-to-add

Scroll list panel to any changes made to the currently viewed list

thomastanck committed:
Merge branch 'master' into add-missing-refreshall-command

qjqqyy committed:
test: add tests for SubscribeCommand

thomastanck committed:
Merge pull request #204 from thomastanck/add-missing-refreshall-command

Add missing refreshall command to list context

rlrh committed:
Update StringUtil, StringUtilTest

rlrh committed:
Merge remote-tracking branch 'upstream/master' into util-tests

thomastanck committed:
Add offline command to enable/disable offline mode

thomastanck committed:
Add tests for offline mode

thomastanck committed:
Add offline mode command to user guide

thomastanck committed:
Position enable/disable literal first in String comparison

thomastanck committed:
Add tests for model's offline mode behaviour and offline command

qjqqyy committed:
Merge pull request #206 from qjqqyy/add-sub-tests

add more tests for subscribe

rlrh committed:
Merge branch 'master' into util-tests

epicfailname committed:
Update and add diagrams

rlrh committed:
Merge branch 'master' into update-aboutus

qjqqyy committed:
StringUtil: use more succinct implementation

epicfailname committed:
Update DG

qjqqyy committed:
parser: lift find command to all contexts

epicfailname committed:
Update UG and PPP Jonathan

epicfailname committed:
Merge pull request #213 from CS2103-AY1819S2-W10-1/update-dg-jonathan

Update docs for Jonathan

qjqqyy committed:
tests: move showNoEntry to CommandTestUtil

qjqqyy committed:
tests: add tests for unsub

epicfailname committed:
Update UG diagram to make PP smaller

epicfailname committed:
Remove section in PPP for size req

epicfailname committed:
Test

epicfailname committed:
Merge pull request #214 from CS2103-AY1819S2-W10-1/update-dg-jonathan

Update UG diagrams to make PPP smaller

qjqqyy committed:
Merge pull request #212 from CS2103-AY1819S2-W10-1/add-moar-tests

Lift FindCommand & add more tests

thomastanck committed:
Update Thomas' PPP

thomastanck committed:
Merge remote-tracking branch 'origin/master' into thomas-ppp

thomastanck committed:
Merge pull request #215 from thomastanck/thomas-ppp

Update Thomas' PPP

thomastanck committed:
Merge pull request #199 from thomastanck/remove-sample-data

Remove sample data on startup

thomastanck committed:
Merge remote-tracking branch 'origin/master' into disable-offline-mode

thomastanck committed:
Merge remote-tracking branch 'origin/master' into disable-offline-mode

thomastanck committed:
Merge pull request #210 from thomastanck/disable-offline-mode

Add offline command to enable/disable offline mode

thomastanck committed:
Remove [since xxx] and [updated xxx] tags

epicfailname committed:
Add deleteall command for Archives and List context

thomastanck committed:
Update quick tour

epicfailname committed:
Add tests for deleteall command

rlrh committed:
Merge pull request #201 from rlrh/update-aboutus

AboutUs: Change team info, add rlrh portfolio link

rlrh committed:
Merge pull request #209 from rlrh/util-tests

Fix StringUtil, add more tests in StringUtilTest

qjqqyy committed:
update qjqqyy ppp & dg parts

thomastanck committed:
Add all commands to User Guide

qjqqyy committed:
use newer pic for DisplayEntryBookSeqDiag

epicfailname committed:
Merge pull request #218 from CS2103-AY1819S2-W10-1/delete-all

Add deleteall command for Archives and List context

thomastanck committed:
Add tag for offline mode command

thomastanck committed:
Update refresh doc and fix xrefs

thomastanck committed:
Merge pull request #217 from thomastanck/update-ug

Update User Guide

qjqqyy committed:
Merge branch 'master' into update-docs

qjqqyy committed:
UG: add images for sub command

epicfailname committed:
Update UG UI image and description

rlrh committed:
First update of rlrh PPP

thomastanck committed:
Add section on async in Network

thomastanck committed:
Update storage class diagram

qjqqyy committed:
UG: fix typo

thomastanck committed:
Update storage component documentation

thomastanck committed:
Update UserGuide.adoc

thomastanck committed:
UG: Fix typo in find command title

thomastanck committed:
Merge pull request #224 from thomastanck/update-ug

UG: Fix typo in find command title

thomastanck committed:
Merge pull request #222 from CS2103-AY1819S2-W10-1/thomastanck-patch-1

Update UserGuide.adoc

thomastanck committed:
Merge pull request #221 from thomastanck/update-dg

Update Developer Guide

qjqqyy committed:
Merge pull request #219 from CS2103-AY1819S2-W10-1/update-docs

update @qjqqyy related docs [1/2]

epicfailname committed:
Add command summary section

thomastanck committed:
UG: update archiveall, TT PPP: remove section

epicfailname committed:
Merge branch 'master' into add-ug-quickstart-image

thomastanck committed:
Update gradle task to generate puml diagrams

epicfailname committed:
Fix bugs

thomastanck committed:
Add TestDiagram.png

thomastanck committed:
Finish section on Offline mode

rlrh committed:
Update Reader View section in DG

epicfailname committed:
Add tip for add command (unique entry/link requirement)

qjqqyy committed:
update

epicfailname committed:
Merge pull request #220 from CS2103-AY1819S2-W10-1/add-ug-quickstart-image

Update UG UI image and add Command Summary

qjqqyy committed:
Merge pull request #225 from thomastanck/update-dg

Update Developer Guide

qjqqyy committed:
Merge branch 'master' into update-docs

rlrh committed:
Merge pull request #226 from rlrh/dg

Update Reader View section in DG

thomastanck committed:
Add activity diagram to docs

rlrh committed:
Update rlrh PP, UG tags

thomastanck committed:
Add offline mode to thomas PPP

qjqqyy committed:
finish up @qjqqyy DG

rlrh committed:
Merge branch 'master' into rlrh-ppp

rlrh committed:
Merge pull request #228 from rlrh/rlrh-ppp

Update rlrh PPP, tag rlrh features in UG

rlrh committed:
Add FAQ

thomastanck committed:
Make everything fit in PPP

qjqqyy committed:
aa

qjqqyy committed:
Merge pull request #227 from CS2103-AY1819S2-W10-1/update-docs

update more @qjqqyy related docs

thomastanck committed:
Add captions

thomastanck committed:
Merge pull request #230 from thomastanck/update-dg

Update DG and PPP

rlrh committed:
Merge pull request #229 from rlrh/ug

Add FAQ
